File: java-code/nonRepRandom.java
import java.util.ArrayList;
import java.util.Random;

public class nonRepRandom{

    private ArrayList<Integer> nums;
    private int max; // specifies the range of numbers [0...max]

    nonRepRandom(int n){
        max = n;
        nums = new ArrayList<Integer>();
        for(int i = 0; i <= max; i++){
            nums.add(i);
        }
    }

    // generates a number between 0 and max, that hasn't been chosen before, if all numbers have been chosen, refill
    public Integer generateNum(){
        if(nums.isEmpty()){
            for(int i = 0; i <= max; i++){
                nums.add(i);
            }
        }
        Random rand = new Random();
        int index = rand.nextInt(nums.size());
        Integer output = nums.get(index);
        nums.remove(index);
        return output;
    }

}

File: python-code/NonRepRandom.py
import random

class NonRepRandom:
    def __init__(self, max):
        self.max = max
        nums = list(range(0,max+1))

    def generateNum():
        if nums.len == 0 :
            nums = list(range(0,max+1))  # repopulates list if empyy
        return nums.pop(random.randint(0,nums.len + 1)) # remove and return a random element from list
        


